frontstage/views/surveys.py

```diff
@@ -1,37 +1,164 @@
+        elif response.status_code == 400:
+            logger.info('Enrolment code already used')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 200
+        elif response.status_code != 200:
+            logger.error('Failed to submit enrolment code')
+            raise ApiError(response)
+
+        encrypted_enrolment_code = cryptographer.encrypt(enrolment_code.encode()).decode()
+        logger.info('Successful enrolment code submitted')
+        return redirect(url_for('surveys_bp.survey_confirm_organisation',
+                                encrypted_enrolment_code=encrypted_enrolment_code,
+                                _external=True,
+                                _scheme=getenv('SCHEME', 'http')))
+
+    elif request.method == 'POST' and not form.validate():
+        logger.info('Invalid character length, must be 12 characters')
+        template_data = {"error": {"type": "failed"}}
+        return render_template('surveys/surveys-add.html', _theme='default',
+                               form=form, data=template_data)
+
+    return render_template('surveys/surveys-add.html', _theme='default',
+                           form=form, data={"error": {}})
+
+
+@surveys_bp.route('/add-survey/confirm-organisation-survey', methods=['GET'])
+@jwt_authorization(request)
+def survey_confirm_organisation(session):
+
+    if request.method == 'GET':
+        # Get and decrypt enrolment code
+        encrypted_enrolment_code = request.args.get('encrypted_enrolment_code', None)
+        enrolment_code = cryptographer.decrypt(encrypted_enrolment_code.encode()).decode()
+
+        logger.info('Attempting to retrieve data for confirm add organisation/survey page')
+        response = api_call('POST', app.config['CONFIRM_ADD_ORGANISATION_SURVEY'],
+                            json={'enrolment_code': enrolment_code})
+
+        if response.status_code != 200:
+            logger.error('Failed to retrieve data for confirm add organisation/survey page')
+            raise ApiError(response)
+
+        response_json = json.loads(response.text)
+        logger.info('Successfully retrieved data for confirm add organisation/survey page')
+        return render_template('surveys/surveys-confirm-organisation.html',
+                               _theme='default',
+                               enrolment_code=enrolment_code,
+                               encrypted_enrolment_code=encrypted_enrolment_code,
+                               organisation_name=response_json['organisation_name'],
+                               survey_name=response_json['survey_name'])
+
+
+@surveys_bp.route('/add-survey/add-survey-submit', methods=['GET'])
+@jwt_authorization(request)
+def add_survey_submit(session):
+    party_id = session['party_id']
+
+    logger.info('Assigning new survey to a user')
```

Comment:
For most of these log lines we have access to the `party_id` it'd probably be good practice to include it wherever we can